### DISCUSSION TITLE
Catalogue Response

### DISCUSSION THREAD

**Discussion by idolum** [eprocurementontology/eprocurementontology#524] [Ideas]
As part of the catalogue choreographies defined in prEN 17015-1 by CEN/TC 440/WG 5, a reponse document to a sent catalogue is needed.

## User Stories

A catalogue provider sends a catalogue to the catalogue receiver. The catalogue is validated by the catalogue receiver and accepted. The catalogue receiver sends a response to indicated to the catalogue provider that the catalogue was accepted and in operation in the ordering system on the buying side.

Another catalogue provider sends a catalogue to the catalogue receiver. The catalogue is validated by the catalogue receiver and was rejected. To reaons why the catalogue was rejected is sent in the response. The reasons for the rejection

1. are provided for the whole catalgue
2. are provided for some of the catalogue lines

## Examples

* https://test-docs.peppol.eu/poacc/upgrade-3/syntax/CatalogueResponse/tree/

## Proposal

Based on the current draft and Peppol:

![image](https://github.com/OP-TED/ePO/assets/1178743/3a0481a5-3ba8-467a-9329-7b01662fcac2)

* Peppol uses a code list UN/Cefact 4343 Response type code (AP= Accepted, RE = Rejected) for the response
  - https://service.unece.org/trade/uncefact/vocabulary/uncl4343/

**Comment by idolum** [comment]
Revised the proposal

**Comment by andreea-pasare** [comment]
![image](https://github.com/OP-TED/ePO/assets/88081600/9e24c9fb-3275-411d-bd95-1a7976186c74)

@idolum please also provide definitions for the newly added concepts.

**Comment by idolum** [comment]
@andreea-pasare Here are my proposals for the missing definitions

epo-cat:CatalogueResponse
> A formal reply from the Catalogue Receiver to the Catalogue Provider stating the Catalogue Receiver's ability to apply the catalogue for placing orders.
> Additionall Information:
> The Catalogue Response is used to inform on the acceptance or rejection of a Catalogue.

epo-cat:CatalogueResponseLine
> Details concerning the Catalogue Receiver’s ability to apply a given Catalogue Line for placing orders.
> Additional information:
> The Catalogue Response Line is used to inform on the acceptance or rejection of a Catalogue Line.

These definitions are aligned with the definitions of the similar concepts of the order response. So, they should fit.

Furthermore, I would like to add an attribute to epo-cat:CatalogueResponse and epo-cat:CatalogueResponseLine:

epo-cat:hasResponseDescription [0..1]
> Response clarification of the Catalogue Receiver’s decision.

Also aligned with the order response. Here, also epo-ord:hasResponseDescription can be reused instead. But in this case, the definition should be more generic.

**Comment by idolum** [comment]
@andreea-pasare @AchillesDougalis Have my last remarks considered? Or do you have any remark on my proposed definitions?

**Reply by andreea-pasare** [reply]
@idolum , we added the proposed definitions for for `epo-cat:CatalogueResponse` and `epo-cat:CatalogueResponseLine` and the attribute `epo-ord:hasResponseDescription` was added as depicted below:
![image](https://github.com/user-attachments/assets/28a69bc4-e9b5-4a3a-9656-ec780fd4addd)
Also, a new ticket to track the alignment of the definition for `epo-ord:hasResponseDescription` was created [#730](https://github.com/OP-TED/ePO/issues/730)
